### DISCUSSION TITLE
SDIO Tuning issue running latest Kernel in Buildroot

### DISCUSSION THREAD

**Discussion by josh25219** [polarfire-soc/polarfire-soc-documentation#131] [Linux drivers for PolarFire Soc] [answered]
Hi,
We use SDIO to communicate with our wifi card.    We used to use buildroot release 2021.11 which used kernel version 5.12.1.   When we upgraded to release 2022.05, we have a SDIO Tuning issue.
All codes are based on:
 [https://github.com/polarfire-soc/polarfire-soc-buildroot-sdk](https://urldefense.com/v3/__https:/github.com/polarfire-soc/polarfire-soc-buildroot-sdk__;!!Hd04JgDbGoDkW7xTBQ!3jQGSvs-uWFB5be_3i6gplD8HI5dtqogPQZz3IxNeKwBOLbpnEHNTDsOU2CrFu_7UQkEKagAbhfMBdyboUE9c-_Km8xqBwMC$) 
[https://github.com/linux4microchip/linux/blob/29c2077400e2614da14e8e5b31f7bec041f3f4be/drivers/mmc/host/sdhci-cadence.c](https://urldefense.com/v3/__https:/github.com/linux4microchip/linux/blob/29c2077400e2614da14e8e5b31f7bec041f3f4be/drivers/mmc/host/sdhci-cadence.c__;!!Hd04JgDbGoDkW7xTBQ!3jQGSvs-uWFB5be_3i6gplD8HI5dtqogPQZz3IxNeKwBOLbpnEHNTDsOU2CrFu_7UQkEKagAbhfMBdyboUE9c-_Km9mu1auy$)

The new kernel casued SDIO tuning failure. The reason is that the driver drivers/mmc/host/sdhci-cadence.c added .platform_execute_tuning = sdhci_cdns_execute_tuning, When sdhci_execute_tuning is called, it will use callback function and call sdhci_cdns_execute_tuning and get -EIO error with ""no tuning point found". Removing .platform_execute_tuning = sdhci_cdns_execute_tuning seems to fix the issue.

Is this expected?

Thanks,
Josh

**Comment by hughbreslin** [comment, marked as answer]
Hi @josh25219 

Can I check that when you remove `.platform_execute_tuning = sdhci_cdns_execute_tuning` your card comes up and works as expected? If so this is expected.

**Reply by josh25219** [reply]
Hi Hugh, 

Yes, removing that line, then things work as expected.   I just wanted to make sure that was an expected issue and appropriate work-around.  Is tuning not required?

Thanks
Josh

**Reply by hughbreslin** [reply]
Cool 🙂 

The tuning thats running there is software tuning as far as I know, which isn't required by your set up as that tuning should be done by the hardware.
